firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/ImpressionStorageClient.java

```diff
@@ -84,7 +85,11 @@ private void clearInMemCache() {
   }
 
   /** Returns {@code Single.just(true)} if the campaign has been impressed */
-  public Single<Boolean> isImpressed(String campaignId) {
+  public Single<Boolean> isImpressed(CampaignProto.ThickContent content) {
+    String campaignId =
```

Comment:
nit: In experimental payload case, this is technically an `experimentId`, right? Should add that as a comment / clarification and / or rename this variable?

Reply:
Yeah, throughout the rest of the code we call this a campaignId and basically what it really is is a unique message identifier. In reality it should probably just be ID. it doesn't even matter that it's a campaignId or an experimentId just that it uniquely identifies this message. so I've just left it was campaignId.